src/cmdhandler/command_handler.cpp

```diff
@@ -191,18 +197,38 @@ std::string command_handler::state_system(const std::vector<std::string>& tokens
         std::string state_to_delete = tokens[2];
 
         //if value is 'current', clear out current state
-        //if not, attempt to delete save state's file
+        //if not, attempt to delete given save state's file
         if(state_to_delete == "current"){
             current_state.robots.clear();
             current_state.collectors.clear();
             return "current state has been cleared";
         }else{
-            if(remove(state_to_delete.c_str()) != 0){
+            if(remove(("states/"+state_to_delete).c_str()) != 0){
                 return "saved state '"+state_to_delete+"' does not exist";
             }else{
                 return "state '"+state_to_delete+"' has been removed";
             }
         }
+    }else if(tokens[0] == "list"){
+        std::string response = "Saved states:";
+
+        DIR *states_dir = opendir("states/");
```

Comment:
We're using C++17 for this project, so instead of getting the available files in a directory like this, you can use the `std::filesystem::directory_iterator` that is now a part of the standard library since C++17: https://en.cppreference.com/w/cpp/filesystem/directory_iterator